include/sparrow/array.hpp

```diff
@@ -47,9 +55,22 @@ namespace sparrow
         SPARROW_API size_type size() const;
         SPARROW_API const_reference operator[](size_type) const;
 
+        SPARROW_API cloning_ptr<array_wrapper> && extract_array_wrapper() &&;
+
     private:
 
         cloning_ptr<array_wrapper> p_array = nullptr;
+
+        friend detail::array_access;
     };
+
+    template<class ARRAY_TYPE>
+    requires std::is_rvalue_reference_v<ARRAY_TYPE&&>
+    array::array(ARRAY_TYPE&& array)
+    {   
+        auto storage = detail::array_access::extract_arrow_proxy(std::move(array));
+        p_array = array_factory(std::move(storage));
+    }
```

Comment:
Since ARRAY_TYPE is known, maybe something like:
```cpp
    template<class ARRAY_TYPE>
    requires std::is_rvalue_reference_v<ARRAY_TYPE&&>
    array::array(ARRAY_TYPE&& array)
        : p_array(new array_wrapper_impl<ARRAY_TYPE>(std::move(array)))
    {
    }
```

Would be enough? It avoids extracting the proxy and going through the factory.